packages/react-dom/src/__tests__/ReactMount-test.js

```diff
@@ -173,12 +482,632 @@ describe('ReactMount', () => {
         div,
       ),
     ).toWarnDev(
-      'Server: "This markup contains an nbsp entity:   server text" ' +
-        'Client: "This markup contains an nbsp entity:   client text"',
+      'Warning: Text content did not match. ' +
+        'Server: "This markup contains an nbsp entity:   server text" ' +
+        'Client: "This markup contains an nbsp entity:   client text"\n' +
+        '    in div (at **)',
+    );
+  });
+
+  it('should warn when a hydrated element has extra props with non-null values', () => {
+    // See fixtures/ssr: ssr-warnForPropDifference
+
+    const div = document.createElement('div');
+    const markup = ReactDOMServer.renderToString(
+      <div>
+        <em>SSRMismatchTest default text</em>
+      </div>,
+    );
+    div.innerHTML = markup;
+
+    expect(() =>
+      ReactDOM.hydrate(
+        <div data-ssr-extra-prop={true} data-ssr-extra-prop-2={true}>
+          <em>SSRMismatchTest default text</em>
+        </div>,
+        div,
+      ),
+    ).toWarnDev(
+      'Warning: Prop `data-ssr-extra-prop` did not match. ' +
+        'Server: null ' +
+        'Client: true\n' +
```

Comment:
Change to the warning message compared to PR #12063:
- the values are not converted to strings before being printed with `JSON.stringify`, so they no longer appear in quotes if they are not strings